src/sparseml/transformers/utils/initializers.py

```diff
@@ -0,0 +1,203 @@
+
+from sparseml.optim import parse_recipe_variables
+from sparseml.transformers.sparsification import Trainer
+from sparseml.transformers.utils.helpers import TaskNames
+from sparseml.transformers.utils.load_task_model import load_task_model
+
+
+__all__ = [
+    "initialize_model",
+    "initialize_tokenizer",
+    "initialize_trainer",
+    "initialize_config",
+    "resolve_sequence_length",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# TODO: Let's not take this for granted. We should be able to
+# device argument
+@dataclass
+class ForceCPUTrainingArguments(TrainingArguments):
+    @property
+    def place_model_on_device(self):
+        # The property governs whether to automatically place
+        # the model on the device. Setting to False ensures that the
+        # model remains in CPU during ONNX export
+        return False
+
+
+def initialize_config(
+    model_path: Union[str, Path], trust_remote_code: bool = False, **config_args
+) -> AutoConfig:
+    """
+    Initialize a config from a given path
+
+    :param model_path: the path to the model to load
+    :param trust_remote_code: True to trust remote code when loading the model,
+        False otherwise
+    :param config_args: additional arguments to pass to the config
+    :return: the loaded config
+    """
+    config = AutoConfig.from_pretrained(
+        model_path,
+        trust_remote_code=trust_remote_code,
+        **config_args,
+    )
+    return config
+
+
+def initialize_tokenizer(
+    model_path: Union[str, Path], sequence_length: int, task: str
+) -> AutoTokenizer:
+    """
+    Initialize a tokenizer from a given path
+
+    :param model_path: the path to the model to load
+    :param sequence_length: the sequence length to use for the tokenizer
+    :param task: the task to load the tokenizer for
+    :return: the loaded tokenizer
+    """
+
+    tokenizer = AutoTokenizer.from_pretrained(
+        model_path, model_max_length=sequence_length
+    )
+    if task in TaskNames.text_generation.value:
+        tokenizer.pad_token = tokenizer.eos_token
+    return tokenizer
+
+
+def initialize_model(
+    model_path: Union[str, Path],
+    task: str,
+    config: AutoConfig,
+    trust_remote_code: bool = False,
+) -> AutoModel:
+    """
+    Initialize a model from a given path
+
+    :param model_path: the path to the model to load
+    :param task: the task to load the model for
+    :param config: the config to use for the model
+    :param trust_remote_code: True to trust remote code when loading the model,
+        False otherwise
+    :return: the loaded model
+    """
+    model = load_task_model(
+        task=task,
+        model_path=model_path,
+        config=config,
+        trust_remote_code=trust_remote_code,
+    )
+    return model
+
+
+def initialize_trainer(
+    model: AutoModel,
+    model_path: Union[str, Path],
+    validation_dataset: Optional[Any] = None,
+) -> Trainer:
+    """
+    Initialize a trainer
```

Comment:
doc string incomplete - should mention if this applies recipe structure, etc

additionally, what is the plan to migrate post sparseml v2 update?

Reply:
as mentioned in the follow-up PR, I will sync with @Satrat to learn about the most recent state of sparseml v2 and will update the logic. Let's not make it a blocker for landing this PR though.